Tradução capítulo 2

GSFZamai opened the pull request:
Concluída a tradução dos subtítulos

GSFZamai committed:
Tradução dos sub-títulos

Iniciando pela tradução dos subtítulos

GSFZamai committed:
Update 02_program_structure.md

GSFZamai committed:
Iniciando tradução expressões e declarações

GSFZamai committed:
Tradução do capítulo

Iniciada a tradução do capítulo.

GSFZamai committed:
Merge pull request #2 from GSFZamai/GSFZamai-patch-1

Traduzido até a linha 20.

joaostein commented:
Fala, @GSFZamai. Para facilitar a revisão do capítulo 02, vamos manter um único _pull request_. Basta continuar enviando as alterações por aqui que, após fazermos a tradução e revisão, fazemos o _squash_ e _merge_ em um único _commit_.

Quando  estiver pronto para revisão, poderia, por gentileza, me dar um ping avisando?

GSFZamai commented:
Bom dia, @joaostein, beleza!
Pode xiexar que mando sim, um abraço!

GSFZamai committed:
Tópico expressões e declarações

Traduzido até a linha 41.

GSFZamai committed:
Finalizado

Traduzido da linha 41 até a linha 53

GSFZamai committed:
Merge pull request #3 from GSFZamai/expressoes_e_declaracoes

Tópico expressões e declarações

GSFZamai committed:
Tradução tópico Ligações/variáveis

Concluída tradução do tópico Ligações/variáveis

GSFZamai committed:
Tradução de dois tópicos

Concluída as traduções para os tópicos:
Nomes das ligações/variáveis e;
O Ambiente

GSFZamai committed:
Traduzido funções e função console.log

Traduzido os tópicos:
Funções;
A função console.log

GSFZamai committed:
Traduzido tópico Retorno de valores

Finalizado a tradução do tópico:
Retorno de Valores.

GSFZamai committed:
Tradução do Tópico Controle de fluxo

Finalizada a tradução do tópico Controle de fluxo.

GSFZamai committed:
Tradução do capitulo Execução condicional

Finalizada a tradução do capítulo Execução condicional

GSFZamai committed:
Traduzido tópico Loops While e Do

Finalizada a tradução do tópico Loops While e Do

GSFZamai committed:
Traduzido 2 tópicos

Finalizado a tradução dos tópicos:
Identação do código.
For loops.

GSFZamai committed:
Traduzido dois tópicos

Traduzido os tópicos Saindo de um Loop e Atualizando ligações resumidamente

GSFZamai committed:
Traduzido dois tópicos

Clocluída as traduções dos tópicos:
- Enviando um valor com *switch*
- Capitalização

GSFZamai committed:
Traduzido até a linha 950

Traduzido até a linha 950, no subtópico FizzBuzz.

GSFZamai committed:
Tradução Capítulo 2

Finalizada a tradução do capítulo 2.

GSFZamai commented:
Bom dia, @joaostein  
Finalizada a tradução do capítulo, pronto para revisão.

joaostein commented on 02_program_structure.md:
```diff
@@ -687,58 +655,56 @@ for (let current = 20; ; current = current + 1) {
-an end condition that always produces `true`, your program would get
-stuck in an _((infinite loop))_. A program stuck in an infinite loop
-will never finish running, which is usually a bad thing.
+Se você remover a declaração `break` ou você acidentamente escreveu
```
acidentamente -> acidentalmente

joaostein commented on 02_program_structure.md:
```diff
@@ -687,58 +655,56 @@ for (let current = 20; ; current = current + 1) {
-you'll usually be asked whether you want to stop the script after a
-few seconds. If that fails, you will have to close the tab that you're
-working in, or on some browsers close your whole browser, to recover.
+Se você crear um loop infinito em um dos exemplos nessas páginas,
```
crear -> criar
loop -> sempre em itálico

joaostein commented on 02_program_structure.md:
```diff
@@ -687,58 +655,56 @@ for (let current = 20; ; current = current + 1) {
-few seconds. If that fails, you will have to close the tab that you're
-working in, or on some browsers close your whole browser, to recover.
+Se você crear um loop infinito em um dos exemplos nessas páginas,
+você normalmente será questionado se quer finalizar o script após alguns segundos.
```
script -> _script_ (itálico)

joaostein commented on 02_program_structure.md:
```diff
@@ -687,58 +655,56 @@ for (let current = 20; ; current = current + 1) {
+A palavra-chave `continue` é similar ao `break`, ao influenciar o progresso
+de um loop. Quando `continue` é encontrado no corpo de um loop,
+comanda um salto para fora do corpo e continua com a próxima iteração do loop.
 control jumps out of the body and continues with the loop's next
```
linha em inglês

joaostein commented on 02_program_structure.md:
```diff
@@ -687,58 +655,56 @@ for (let current = 20; ; current = current + 1) {
 
-Especially when looping, a program often needs to "update" a binding
-to hold a value based on that binding's previous value.
+Especialmente quando ao usar um loop, o programa com frequência necessita "atualizar" a ligação
```
ao usar -> usar

joaostein commented on 02_program_structure.md:
````diff
@@ -687,58 +655,56 @@ for (let current = 20; ; current = current + 1) {
 ```
-
-JavaScript provides a shortcut for this.
+JavaScript fornece um atalho para isso
````
isso -> isso.

joaostein commented on 02_program_structure.md:
````diff
@@ -687,58 +655,56 @@ for (let current = 20; ; current = current + 1) {
 counter += 1;
 ```
+Atalhos similares funcionam para muitos outros operadores, assim como `result *= 2`
+para dobrar `result` ou `conter -= 1` para contar regressivamente.
````
`conter -= 1` -> `counter -= 1`

joaostein replied:
`conter -= 1` -> `counter -= 1`

joaostein commented on 02_program_structure.md:
```diff
@@ -748,14 +714,14 @@ for (let number = 0; number <= 12; number += 2) {
+equivalente, mais curta: `counter++` e `counter--`.
 
-## Dispatching on a value with switch
+## Enviando um valor com *switch*
```
usar `_` para itálico

joaostein commented on 02_program_structure.md:
```diff
@@ -748,14 +714,14 @@ for (let number = 0; number <= 12; number += 2) {
 {{index [syntax, statement], "conditional execution", dispatch, ["if keyword", chaining]}}
 
-It is not uncommon for code to look like this:
+Não é incomum um código se parece com isso:
```
parece -> parecer

joaostein commented on 02_program_structure.md:
```diff
@@ -766,11 +732,11 @@ else defaultAction();
-JavaScript uses for this (which it inherited from the C/Java line of
-programming languages) is somewhat awkward—a chain of `if` statements
-may look better. Here is an example:
+Exsite uma construção chamada `switch` que tem a inteção de expressar
```
Exsite -> Existe
inteção -> intenção

joaostein commented on 02_program_structure.md:
```diff
@@ -766,11 +732,11 @@ else defaultAction();
-programming languages) is somewhat awkward—a chain of `if` statements
-may look better. Here is an example:
+Exsite uma construção chamada `switch` que tem a inteção de expressar
+o envio de uma forma mai direta. Infelizmente, a sintaxe que o JavaScritp usa para isso
```
mai -> mais
JavaScritp -> JavaScript

joaostein commented on 02_program_structure.md:
```diff
@@ -790,25 +756,22 @@ switch (prompt("What is the weather like?")) {
+assim como o caso `sunny` no exemplo, ele pode ser usado para compartilhar algum código entre _"cases"_
+(ele retorna _"going outside"_ para ambos os climas _"sunny"_ e _"cloudy"_).
+Mas cuidado-é fácil se esquecer de tal `break`, o que irá fazer o programa executar código que você
+não deseja que exetcute.
```
exetcute -> execute

joaostein commented on 02_program_structure.md:
```diff
@@ -790,25 +756,22 @@ switch (prompt("What is the weather like?")) {
-multiple words to clearly describe what the binding represents. These
-are pretty much your choices for writing a binding name with several
-words in it:
+Nomes de ligações não podem conter espaços, mesmo assim usar multiplas palavras
```
multiplas -> múltiplas

joaostein commented on 02_program_structure.md:
```diff
@@ -790,25 +756,22 @@ switch (prompt("What is the weather like?")) {
-words in it:
+Nomes de ligações não podem conter espaços, mesmo assim usar multiplas palavras
+é um facilitador para descrever claramente o que as ligações representam.
+Isso é claramente uma decisão sua nomear uma ligação com várias palarvas:
```
palarvas -> palavras

joaostein commented on 02_program_structure.md:
```diff
@@ -819,38 +782,40 @@ fuzzyLittleTurtle
 
-The first style can be hard to read. I rather like the look of the
+O primeiro estilo pode tornar díficil de ler. Eu particularmente 
+prefiro o visual dos underlines, embora esse estilo seja penoso para digitat.
```
underlines -> itálico
digitat -> digitar

joaostein commented on 02_program_structure.md:
```diff
@@ -819,38 +782,40 @@ fuzzyLittleTurtle
+prefiro o visual dos underlines, embora esse estilo seja penoso para digitat.
+As funções ((padrões)) JavaScript, e a maior parte dos programadores JavaScript,
+seguem o último estilo-eles 
 underscores, though that style is a little painful to type. The
```
linha em inglês

joaostein commented on 02_program_structure.md:
```diff
@@ -819,38 +782,40 @@ fuzzyLittleTurtle
-?](object#constructors). For now, the important thing is not
-to be bothered by this apparent lack of ((consistency)).
+Em poucos casos, como a função `Number`, a primeira letra de uma ligação também é maiúscula.
+Isso é feito para marcar essa função como contrutora. O que é uma função construtora ficará mais claro no
```
contrutora -> construtora

joaostein commented on 02_program_structure.md:
```diff
@@ -819,38 +782,40 @@ fuzzyLittleTurtle
-characters (`//`) and then the comment text after it.
+Um comentário é um pedaço de texto que faz parte do programa mas é
+completamente ignorado pelo computador. JavaScript possui duas formas de escrever
+comentários. Escrever um comentário se única linha, você pode usar dois carácteres de barra
```
Escrever um comentário se única linha -> Para escrever um comentário de linha única

joaostein commented on 02_program_structure.md:
```diff
@@ -819,38 +782,40 @@ fuzzyLittleTurtle
+Um comentário é um pedaço de texto que faz parte do programa mas é
+completamente ignorado pelo computador. JavaScript possui duas formas de escrever
+comentários. Escrever um comentário se única linha, você pode usar dois carácteres de barra
+(`//`) e então o texto de comentário após elas.
```
e então o texto -> e, então, o texto

joaostein commented on 02_program_structure.md:
```diff
@@ -880,51 +845,47 @@ information about a file or a chunk of program.
-expressions.
+Você sabe agora que um programa é construido por declarações, as quais
+algumas vezes possuem mais declarações. Declarações tendem a conter
+expressões, os quais podem ser contruidos de expressões menores.
```
os quais -> as quais
contruidos -> construídas

joaostein commented on 02_program_structure.md:
```diff
@@ -880,51 +845,47 @@ information about a file or a chunk of program.
-executed from top to bottom. You can introduce disturbances in the
-flow of control by using conditional (`if`, `else`, and `switch`) and
-looping (`while`, `do`, and `for`) statements.
+Colocar declarações após outra declaração the dá um programa que é
```
the -> lhe / te

joaostein commented on 02_program_structure.md:
```diff
@@ -880,51 +845,47 @@ information about a file or a chunk of program.
-flow of control by using conditional (`if`, `else`, and `switch`) and
-looping (`while`, `do`, and `for`) statements.
+Colocar declarações após outra declaração the dá um programa que é
+executado do topo para baixo. Você pode introduzir alterações no clontrole de fluxo
```
clontrole -> controle

joaostein commented on 02_program_structure.md:
```diff
@@ -880,51 +845,47 @@ information about a file or a chunk of program.
-looping (`while`, `do`, and `for`) statements.
+Colocar declarações após outra declaração the dá um programa que é
+executado do topo para baixo. Você pode introduzir alterações no clontrole de fluxo
+ao utilizar declarações condicionais (`if`, `else`, e `switch`) e loops (`while`, `do`, e `for`)
```
loops -> itálico

joaostein commented on 02_program_structure.md:
```diff
@@ -880,51 +845,47 @@ information about a file or a chunk of program.
-useful for tracking state in your program. The environment is the set
-of bindings that are defined. JavaScript systems always put a number
-of useful standard bindings into your environment.
+Ligações podem ser usadas para arquivar pedaços de dados sob nome, e elas são
```
sob nome -> em um nome

joaostein commented on 02_program_structure.md:
```diff
@@ -880,51 +845,47 @@ information about a file or a chunk of program.
-of useful standard bindings into your environment.
+Ligações podem ser usadas para arquivar pedaços de dados sob nome, e elas são
+úteis para rastrear estados em seu programa. O ambiente é o conjunto de ligações
+que são definidas. Sistemas JavaScript sempre coloca um número de ligações padrões úteis
```
coloca -> colocam

joaostein commented on 02_program_structure.md:
```diff
@@ -880,51 +845,47 @@ information about a file or a chunk of program.
+[após o exercício]{if interactive}[no [fim do livro](dicas)]{if book}. 
+Soluções completas para os exercícios não estão inclusas nesse livro, mas podem ser encontradas
+online no [_https://eloquentjavascript.net/code_](https://eloquentjavascript.net/code#2).
+Se você deseja aprender algo dos exercícios, Eu recomendo consultar as soluções
```
Eu -> eu

joaostein commented on 02_program_structure.md:
```diff
@@ -880,51 +845,47 @@ information about a file or a chunk of program.
+com força e por tempo suficiente a ponto de ter uma leve dor de cabeça.
 
-### Looping a triangle
+### Triângulo com Loop
```
Loop -> itálico

joaostein commented on 02_program_structure.md:
```diff
@@ -937,9 +898,8 @@ following triangle:
-
-It may be useful to know that you can find the length of a string by
-writing `.length` after it.
+Pode ser útil saber que você pode descobrir o comprimento da string ao escrever
```
string -> itálico

joaostein commented on 02_program_structure.md:
```diff
@@ -949,8 +909,8 @@ console.log(abc.length);
 
-Most exercises contain a piece of code that you can modify to solve
-the exercise. Remember that you can click code blocks to edit them.
+A maioria dos exercícios contém um pedaço de código que você pode modificar para solucionar
```
contém -> contêm

joaostein commented on 02_program_structure.md:
```diff
@@ -949,8 +909,8 @@ console.log(abc.length);
-Most exercises contain a piece of code that you can modify to solve
-the exercise. Remember that you can click code blocks to edit them.
+A maioria dos exercícios contém um pedaço de código que você pode modificar para solucionar
+o exercício. Lembre-se que você pode clicar os blocos e código e editá-los.
```
clicar os blocos e código -> clicar nos blocos de código

joaostein commented on 02_program_structure.md:
```diff
@@ -961,34 +921,33 @@ if}}
-chapter, where the `for` loop was introduced.
+Você pode começar com um programa que imprima os números 1 à 7, que
+você pode derivar ao fazer umas poucas modificações ao 
+you can derive by making a few modifications to the [exemplo de impressão de números ímpares](program_structure#loops) 
```
texto em inglês

joaostein commented on 02_program_structure.md:
```diff
@@ -961,34 +921,33 @@ if}}
-characters. You can go from 1 to 2 by adding 1 (`+= 1`). You can go
-from `"#"` to `"##"` by adding a character (`+= "#"`). Thus, your
-solution can closely follow the number-printing program.
+Agora considere a equivalência entre números e strings de caractéres de cerquilha.
```
strings -> itálico
caractéres -> caracteres

joaostein commented on 02_program_structure.md:
```diff
@@ -961,34 +921,33 @@ if}}
-solution can closely follow the number-printing program.
+Agora considere a equivalência entre números e strings de caractéres de cerquilha.
+Você pode ir de 1 à 2 ao adicionar ' (`+= 1`). Você pode ir de `"#"` para `"##"`
+ao adicionar um caracter (`+= "#"`). Desse modo, sua solução pode estar
```
caracter -> caractere

joaostein commented on 02_program_structure.md:
```diff
@@ -961,34 +921,33 @@ if}}
-weed out a significant percentage of programmer candidates. So if you
-solved it, your labor market value just went up.)
+(Isso é na realidade uma ((questão de entrevista)) que eliminou uma porcentagem  
+significante de candidatos à programadores. Então se você resolvê-lo, seu valor no mercado de trabalho
```
Então se -> Então, se

joaostein commented on 02_program_structure.md:
```diff
@@ -1000,35 +959,33 @@ if}}
-print is a matter of conditional execution. Remember the trick of
-using the remainder (`%`) operator for checking whether a number is
-divisible by another number (has a remainder of zero).
+Exeminar os números é claramente um trabalho para um loop, e selecionar o que
```
Exeminar -> Examinar

joaostein commented on 02_program_structure.md:
```diff
@@ -1000,35 +959,33 @@ if}}
-divisible by another number (has a remainder of zero).
+Exeminar os números é claramente um trabalho para um loop, e selecionar o que
+imprimir é uma questão de execução condicional. Lembre-se do truque de usar o
+operador resto (`%`) para checar se um número é divisivel por outro (possui um resto de zero).
```
divisivel -> divisível

joaostein commented on 02_program_structure.md:
```diff
@@ -1055,26 +1012,25 @@ if}}
 
-You can build the string by starting with an empty one (`""`) and
-repeatedly adding characters. A newline character is written `"\n"`.
+Você pode construir a _string_ ao iniciando por uma vazia (`""`) e
```
ao iniciando -> iniciando

joaostein commented on 02_program_structure.md:
```diff
@@ -1055,26 +1012,25 @@ if}}
-You can build the string by starting with an empty one (`""`) and
-repeatedly adding characters. A newline character is written `"\n"`.
+Você pode construir a _string_ ao iniciando por uma vazia (`""`) e
+repetidamente adicionando characters. Um caracter _newline_ é escrito com `"\n"`.
```
adicionando -> adicionar
characters -> caracteres

joaostein commented on 02_program_structure.md:
```diff
@@ -1055,26 +1012,25 @@ if}}
-outer loop handles the lines, and the inner loop handles the characters
-on a line.
+Para trabalhar com duas ((dimensões)), você irá precisar de um ((loop)) dentro de um loop.
+Coloque chaves em volta do corpo de embos os loops para tornar fácil de identificar
```
embos -> ambos
loop -> itálico (sempre)

joaostein commented on 02_program_structure.md:
```diff
@@ -1055,26 +1012,25 @@ if}}
-on a line.
+Para trabalhar com duas ((dimensões)), você irá precisar de um ((loop)) dentro de um loop.
+Coloque chaves em volta do corpo de embos os loops para tornar fácil de identificar
+onde eles começam e terminam. Tente identar apropriadamente o corpo dos loops.
```
identar -> indentar

joaostein commented on 02_program_structure.md:
```diff
@@ -1055,26 +1012,25 @@ if}}
+Para trabalhar com duas ((dimensões)), você irá precisar de um ((loop)) dentro de um loop.
+Coloque chaves em volta do corpo de embos os loops para tornar fácil de identificar
+onde eles começam e terminam. Tente identar apropriadamente o corpo dos loops.
+A ordem dos loops deve seguir a order que nós contruimos a _string_ 
```
contruimos -> construímos

joaostein commented on 02_program_structure.md:
```diff
@@ -1055,26 +1012,25 @@ if}}
+onde eles começam e terminam. Tente identar apropriadamente o corpo dos loops.
+A ordem dos loops deve seguir a order que nós contruimos a _string_ 
+(linha por lina, esquerda para direita, de cima para baixo).
+Então o loop externo cuida das linhas, e o loop interno cuida dos caractéres da linha.
```
caractéres -> caracteres

joaostein commented on 02_program_structure.md:
```diff
@@ -1055,26 +1012,25 @@ if}}
 
-Terminating a line by adding a newline character must happen after the
-line has been built up, so do this after the inner loop but inside the outer loop.
+Terminando a linha ao adiconar um caracter _newline_ deve acontecer após a
```
caracter -> caractere

joaostein reviewed (changes requested):
valeu @GSFZamai! Adicionei alguns comentários de review.

GSFZamai committed:
Update 02_program_structure.md

GSFZamai committed:
Update 02_program_structure.md

GSFZamai committed:
Update 02_program_structure.md

GSFZamai committed:
Update 02_program_structure.md

GSFZamai committed:
Update 02_program_structure.md

GSFZamai committed:
Update 02_program_structure.md

GSFZamai committed:
Update 02_program_structure.md

GSFZamai reviewed (commented):
Bom dia @joaostein , revisado!

joaostein commented on 02_program_structure.md:
```diff
@@ -1,112 +1,103 @@
+Isso mostra parte da beleza de uma interface baseada em linguagem.
+Expressões podem conter outras expressões de forma similar à como subfrases 
+na linguagem humana são aninhadas.—uma subfrase pode conter sua própria subfrase, e assim por diante. 
+Isso nos permite criar expressões que descrevem cálculos arbitráriamente complexos.
```
arbitráriamente -> arbitrariamente

joaostein commented on 02_program_structure.md:
```diff
@@ -1,112 +1,103 @@
-about the subtleties of missing semicolons.
+Em alguns casos, JavaScript te permite omitir o ponto e vírgula ao final de uma declaração. 
+Em outros casos, ele tem que estar ali, ou a próxima ((linha)) será tratada como parte da mesma declaração. 
+As regras para quando for seguro omití-lo são de alguma forma complexas e passíveis de erro. 
```
omití-lo -> omiti-lo

joaostein commented on 02_program_structure.md:
```diff
@@ -1,112 +1,103 @@
-a binding. It is followed by the name of the binding and, if we want
-to immediately give it a value, by an `=` operator and an expression.
+Essa é a segunda forma de ((declaração)). A palavra especial
+(_((paravra-chave))_) `let` indica que essa declaração irá definir
```
paravra-chave -> palavra-chave

joaostein commented on 02_program_structure.md:
```diff
@@ -116,10 +107,10 @@ console.log(ten * ten);
-them point to a new one.
+Quando uma ligação aponta para um valor, não significa que ela está amarrada àquele valor
+para sempre. O operador `=` pode ser usado a qualquer momento em
+ligações existentes para desconectá-las de seu valor atual e apontá-las
```
desconectá-las -> desconecta-las

joaostein commented on 02_program_structure.md:
```diff
@@ -151,23 +142,23 @@ console.log(luigisDebt);
-When you define a binding without giving it a value, the tentacle has
-nothing to grasp, so it ends in thin air. If you ask for the value of
-an empty binding, you'll get the value `undefined`.
+Quando você define uma ligação sem atribur um valor à ela, o tentáculo
```
atribur -> atribuir

joaostein commented on 02_program_structure.md:
```diff
@@ -178,35 +169,34 @@ console.log(greeting + name);
+O primeiro, `var` (abreviação para "variável"), é a forma como as ligações
+eram declaradas no JavaScript pre-2015. Eu irei precisar como ela difere de `let`
+no [próximo capítulo](functions). Por enquanto,
+se lembre que elas básicamente fazem a mesma coisa, mas nós raramente a usaremos nesse livro
```
básicamente -> basicamente

joaostein commented on 02_program_structure.md:
```diff
@@ -178,35 +169,34 @@ console.log(greeting + name);
+Nomes de ligações podem ser qualquer palavra. Dígitos podem ser parte do nome da ligação
+—`catch22` é um nome válido, por exemplo— mas o nome não deve começar com um dígito.
+O nome de uma ligação pode incluir sinal de dólar (`$`) ou
+_underlines_ (`_`) mas nenhuma outra pontuação ou caractere especial.
```
_underlines_ -> _underscores_ (se está usando inglês, deixa como original)

joaostein commented on 02_program_structure.md:
````diff
@@ -217,36 +207,33 @@ switch this throw true try typeof var void while with yield
 ```
 
 {{index [syntax, error]}}
+Não se preocupe em memorizar essa lista. Quando ao criar uma ligação poduzir o erro  
````
poduzir -> produzir

joaostein commented on 02_program_structure.md:
```diff
@@ -217,36 +207,33 @@ switch this throw true try typeof var void while with yield
+A coleção de ligações e seus valores que existem em um determinado momento
+é chamado de _((ambiente))_. Quando um programa inicia, esse ambiente
+não é vazio. Ele sempre contém ligações que são parte da linguagem
+((padrão)), e na marioria das vezes, ele também possui ligações que fornecem formas
```
marioria -> maioria

joaostein commented on 02_program_structure.md:
```diff
@@ -217,36 +207,33 @@ switch this throw true try typeof var void while with yield
+é chamado de _((ambiente))_. Quando um programa inicia, esse ambiente
+não é vazio. Ele sempre contém ligações que são parte da linguagem
+((padrão)), e na marioria das vezes, ele também possui ligações que fornecem formas
+de interagir com o sistema que o cerca. Por exempo, em um navegador, existem funções para
```
exempo -> exemplo

joaostein commented on 02_program_structure.md:
```diff
@@ -217,36 +207,33 @@ switch this throw true try typeof var void while with yield
+não é vazio. Ele sempre contém ligações que são parte da linguagem
+((padrão)), e na marioria das vezes, ele também possui ligações que fornecem formas
+de interagir com o sistema que o cerca. Por exempo, em um navegador, existem funções para
+interagir com o website ataualmente carregado e para ler as entradas do ((mouse)) e ((teclado)).
```
ataualmente -> atualmente

joaostein commented on 02_program_structure.md:
```diff
@@ -256,38 +243,36 @@ prompt("Enter passcode");
-_((argument))s_. Different functions might need a different number or
-different types of arguments.
+Executar uma função é chamado _invoking_, _calling_, ou _applying_ (no português: _invocar_, _chamar_ ou _aplicar_)
+Você pode chamar uma função ao colocar parentêses após a expressão que produz o valor da expressão.
```
parentêses -> parênteses

joaostein commented on 02_program_structure.md:
```diff
@@ -356,33 +338,31 @@ console.log("Your number is the square root of " +
-execution))_.
+Nem todos os programas são estradas retas. Nós podemos, por exemplo, querer
+criar uma estrada ramificada, onde o progama pega o ramo apropriado
+baseado na situação em que se encontra. Isso é chamado de _((execução condional))_.
```
condional -> condicional

joaostein commented on 02_program_structure.md:
```diff
@@ -392,32 +372,31 @@ if (!Number.isNaN(theNumber)) {
-keyword, between parentheses, followed by the statement to
-execute.
+A palavra-chave `if` executa ou evita uma declaração dependendo do valor de uma
+expressão Booleana. A expressão decisiva é escrita após a palavra-chave, entre parentêses, seguida por
```
parentêses -> parênteses

joaostein commented on 02_program_structure.md:
```diff
@@ -426,10 +405,10 @@ if (1 + 1 == 2) console.log("It's true");
-execution paths.
+Você frequentemente não vai querer um código que apenas execute quando uma condição
+retornar verdadeira, mas também um código que cuida do outro caso. Esse caminho alternativo
+é representado pela segunda flecha do diagrama. Você pode usar a palava-chave `else`, junto com `if`, para criar dois caminhos
```
palava-chave -> palavra-chave

joaostein commented on 02_program_structure.md:
```diff
@@ -443,8 +422,8 @@ if (!Number.isNaN(theNumber)) {
 
-If you have more than two paths to choose from, you can "chain" multiple `if`/`else`
-pairs together. Here's an example:
+Se você possui mais que dois caminhos para escolher, você pode "encadear" múltiplos pares de`if`/`else` juntos.
```
de`if`/`else` -> de `if`/`else` (adicionar espaço)

joaostein commented on 02_program_structure.md:
```diff
@@ -515,28 +490,26 @@ while (number <= 12) {
+Como um exemplo que atualmente faz algo útil, nós podemos agora escrever um
+programa que calcula e exibe o valor de 2^10^ (2 pela 10ª potência). 
+Nós usamos duas ligações: uma para rastrear o resultado e uma para contar
+com qual frequência nós temos que múltiplicar esse resultado por 2. O _loop_ testa
```
múltiplicar -> multiplicar

joaostein commented on 02_program_structure.md:
```diff
@@ -637,23 +604,22 @@ for (let number = 0; number <= 12; number = number + 2) {
+normalmente definindo uma ligação. A segunda parte é a
+((expressão)) que _checa_ se o loop deve continuar. A parte final
+_atualiza_ o estado do loop após cada iteração. Na maioria dos casos,
+isso é mais curto e claro que a contrução `while`.
```
contrução -> construção

joaostein commented on 02_program_structure.md:
```diff
@@ -819,38 +778,37 @@ fuzzyLittleTurtle
-It is not hard to get used to little things like that, and code with
-mixed naming styles can be jarring to read, so we follow this
-((convention)).
+O primeiro estilo pode tornar díficil de ler. Eu particularmente 
```
díficil -> difícil

joaostein commented on 02_program_structure.md:
```diff
@@ -819,38 +778,37 @@ fuzzyLittleTurtle
+As funções ((padrões)) JavaScript, e a maior parte dos programadores JavaScript,
+seguem o último estilo-eles escrevem com a primeira letra maiúscula todas as palavras
+exceto a primeira.
+Não é dificil se acostumar com coisas pequenas como essa, e o código com estilos de nomes
```
dificil -> difícil

joaostein commented on 02_program_structure.md:
```diff
@@ -819,38 +778,37 @@ fuzzyLittleTurtle
-mixed naming styles can be jarring to read, so we follow this
-((convention)).
+O primeiro estilo pode tornar díficil de ler. Eu particularmente 
+prefiro o visual dos _underlines_, embora esse estilo seja penoso para digitar.
```
_underlines_ -> _ underscores_

joaostein commented on 02_program_structure.md:
```diff
@@ -819,38 +778,37 @@ fuzzyLittleTurtle
+seguem o último estilo-eles escrevem com a primeira letra maiúscula todas as palavras
+exceto a primeira.
+Não é dificil se acostumar com coisas pequenas como essa, e o código com estilos de nomes
+mistos podem se tornar duros de ser ler, então seguimos essa ((conveção)).
```
conveção -> convenção

joaostein commented on 02_program_structure.md:
```diff
@@ -880,51 +838,47 @@ information about a file or a chunk of program.
-themselves sometimes contain more statements. Statements tend to
-contain expressions, which themselves can be built out of smaller
-expressions.
+Você sabe agora que um programa é construido por declarações, as quais
```
construido -> construído

joaostein commented on 02_program_structure.md:
```diff
@@ -880,51 +838,47 @@ information about a file or a chunk of program.
-expressions.
+Você sabe agora que um programa é construido por declarações, as quais
+algumas vezes possuem mais declarações. Declarações tendem a conter
+expressões, as quais podem ser costruídas de expressões menores.
```
costruídas -> construídas

joaostein commented on 02_program_structure.md:
```diff
@@ -1055,26 +1004,25 @@ if}}
+Para trabalhar com duas ((dimensões)), você irá precisar de um ((loop)) dentro de um loop.
+Coloque chaves em volta do corpo de ambos os _loops_ para tornar fácil de identificar
+onde eles começam e terminam. Tente indentar apropriadamente o corpo dos loops.
+A ordem dos loops deve seguir a order que nós contruímos a _string_ 
```
contruímos -> construímos
order -> ordem

joaostein commented on 02_program_structure.md:
```diff
@@ -1055,26 +1004,25 @@ if}}
+Coloque chaves em volta do corpo de ambos os _loops_ para tornar fácil de identificar
+onde eles começam e terminam. Tente indentar apropriadamente o corpo dos loops.
+A ordem dos loops deve seguir a order que nós contruímos a _string_ 
+(linha por lina, esquerda para direita, de cima para baixo).
```
lina -> linha

joaostein commented on 02_program_structure.md:
```diff
@@ -1055,26 +1004,25 @@ if}}
-put a space or a hash sign at a given position, you could test whether
-the sum of the two counters is even (`% 2`).
+Você irá precisar de duas ligações para rastrear seu progresso. Para saber se deve
+coloar um espaço ou uma cerquilha na posição dada, você poderia testar se a soma
```
coloar -> colocar

joaostein commented on 02_program_structure.md:
```diff
@@ -1055,26 +1004,25 @@ if}}
 
-Terminating a line by adding a newline character must happen after the
-line has been built up, so do this after the inner loop but inside the outer loop.
+Terminando a linha ao adiconar um caractere _newline_ deve acontecer após a
```
adiconar -> adicionar

joaostein reviewed (changes requested):
@GSFZamai segunda leva de revisão.

GSFZamai committed:
Update 02_program_structure.md

GSFZamai committed:
Concluída revisão

GSFZamai reviewed (commented):
Revisão Concluída

joaostein commented on 02_program_structure.md:
```diff
@@ -664,16 +630,16 @@ console.log(result);
-loop can finish. There is a special statement called `break` that has
-the effect of immediately jumping out of the enclosing loop.
+Ter uma condição do _loop_ produzindo `false` não é a única forma
+que um loop pode sinalizar. Existe uma declaração especial chamada `break` que possui
```
sinalizar -> finalizar

joaostein commented on 02_program_structure.md:
```diff
@@ -687,58 +653,54 @@ for (let current = 20; ; current = current + 1) {
-Using the remainder (`%`) operator is an easy way to test whether a
-number is divisible by another number. If it is, the remainder of
-their division is zero.
+Usando o operador resto (`%`) é uma forma fácil de testar se o número é 
```
Usando -> Usar

joaostein commented on 02_program_structure.md:
```diff
@@ -687,58 +653,54 @@ for (let current = 20; ; current = current + 1) {
-working in, or on some browsers close your whole browser, to recover.
+Se você criar um _loop_ infinito em um dos exemplos nessas páginas,
+você normalmente será questionado se quer finalizar o _script_ após alguns segundos.
+Se isso falhar, você terá que fechar a aba que você está trabalhando,
```
Se isso falhar, você terá que fechar a aba que você está trabalhando ou, em alguns navegadores, fechar todo o navegador para recomeçar.

joaostein commented on 02_program_structure.md:
```diff
@@ -766,11 +728,11 @@ else defaultAction();
+Existe uma construção chamada `switch` que tem a intenção de expressar
+o envio de uma forma mais direta. Infelizmente, a sintaxe que o JavaScript usa para isso
+(que é herdada das linhas de linguagem de programação C/Java)
+é de alguma forma estranha -uma corrente de declarações `if` talvez 
```
`-uma` -> `. Uma`

joaostein commented on 02_program_structure.md:
```diff
@@ -790,25 +752,22 @@ switch (prompt("What is the weather like?")) {
-`break`, which will cause the program to execute code you do not want
-executed.
+Você poderia inserir quantos rótulos de `case` quiser dentro de um bloco aberto por `switch`.  
+O programa irá iniciar a execução no rótulo que corresponde ao valor que foi dado ao `switch`,
```
irá iniciar -> iniciará

joaostein commented on 02_program_structure.md:
```diff
@@ -790,25 +752,22 @@ switch (prompt("What is the weather like?")) {
-executed.
+Você poderia inserir quantos rótulos de `case` quiser dentro de um bloco aberto por `switch`.  
+O programa irá iniciar a execução no rótulo que corresponde ao valor que foi dado ao `switch`,
+ou pelo `default` e nenhuma combinação de valores for encontrada. Ele irá continuar executando,
```
e nenhuma -> se nenhuma

joaostein commented on 02_program_structure.md:
```diff
@@ -790,25 +752,22 @@ switch (prompt("What is the weather like?")) {
+mesmo através de outros rótulos, até encontrar uma declaração `break`. Em alguns casos,
+assim como o caso `sunny` no exemplo, ele pode ser usado para compartilhar algum código entre _"cases"_
+(ele retorna _"going outside"_ para ambos os climas _"sunny"_ e _"cloudy"_).
+Mas cuidado-é fácil se esquecer de tal `break`, o que irá fazer o programa executar código que você
```
`Mas cuidado-` -> `Mas cuidado,`

joaostein commented on 02_program_structure.md:
```diff
@@ -819,38 +778,37 @@ fuzzyLittleTurtle
-It is not hard to get used to little things like that, and code with
-mixed naming styles can be jarring to read, so we follow this
-((convention)).
+O primeiro estilo pode tornar difícil de ler. Eu particularmente 
```
`tornar difícil` -> `ser difícil`

joaostein commented on 02_program_structure.md:
```diff
@@ -819,38 +778,37 @@ fuzzyLittleTurtle
+O primeiro estilo pode tornar difícil de ler. Eu particularmente 
+prefiro o visual dos _underscores_, embora esse estilo seja penoso para digitar.
+As funções ((padrões)) JavaScript, e a maior parte dos programadores JavaScript,
+seguem o último estilo-eles escrevem com a primeira letra maiúscula todas as palavras
```
As funções ((padrões)) JavaScript e a maior parte dos programadores JavaScript seguem o último estilo, escrevendo com a primeira letra maiúscula todas as palavras

joaostein commented on 02_program_structure.md:
```diff
@@ -819,38 +778,37 @@ fuzzyLittleTurtle
-to be bothered by this apparent lack of ((consistency)).
+Em poucos casos, como a função `Number`, a primeira letra de uma ligação também é maiúscula.
+Isso é feito para marcar essa função como construtora. O que é uma função construtora ficará mais claro no
+[Capítulo ](object#constructors). Por enquanto, a coisa mais importante é não se incomodar
```
`[Capítulo ](object#constructors)` -> `[Capítulo ?](object#constructors)`

joaostein commented on 02_program_structure.md:
```diff
@@ -880,51 +838,47 @@ information about a file or a chunk of program.
-executed from top to bottom. You can introduce disturbances in the
-flow of control by using conditional (`if`, `else`, and `switch`) and
-looping (`while`, `do`, and `for`) statements.
+Colocar declarações após outra declaração te dá um programa que é
```
`Colocar declarações após outra declaração` -> `Colocar declarações após as outras`

joaostein commented on 02_program_structure.md:
```diff
@@ -880,51 +838,47 @@ information about a file or a chunk of program.
-can invoke them by writing `functionName(argument1, argument2)`. Such
-a function call is an expression and may produce a value.
+Funções são valores especiais que encapsulam um pedaço de programa. Você
+pode invocá-las ao escrever `functionName(argument1, argument2)`. Assim
```
`Assim` -> `Assim,`

joaostein commented on 02_program_structure.md:
```diff
@@ -961,34 +914,32 @@ if}}
-you can derive by making a few modifications to the [even number
-printing example](program_structure#loops) given earlier in the
-chapter, where the `for` loop was introduced.
+Você pode começar com um programa que imprima os números 1 à 7, que
```
1 à 7 -> 1 a 7

joaostein commented on 02_program_structure.md:
```diff
@@ -961,34 +914,32 @@ if}}
-from `"#"` to `"##"` by adding a character (`+= "#"`). Thus, your
-solution can closely follow the number-printing program.
+Agora considere a equivalência entre números e _strings_ de caracteres de cerquilha.
+Você pode ir de 1 à 2 ao adicionar ' (`+= 1`). Você pode ir de `"#"` para `"##"`
```
1 à 2 -> 1 a 2

joaostein commented on 02_program_structure.md:
```diff
@@ -961,34 +914,32 @@ if}}
-1 to 100, with two exceptions. For numbers divisible by 3, print
-`"Fizz"` instead of the number, and for numbers divisible by 5 (and
-not 3), print `"Buzz"` instead.
+Escreva um programa que usa `console.log` para imprimir todos os números de 1 à 100,
```
1 à 100 -> 1 a 100

joaostein reviewed (changes requested):
@GSFZamai terceiro round de reviews

GSFZamai committed:
Update 02_program_structure.md

GSFZamai committed:
Correções finalizadas

Correções finalizadas.

GSFZamai reviewed (commented):
Correções finalizadas

joaostein commented on 02_program_structure.md:
```diff
@@ -1,112 +1,102 @@
+Ela poderia exibir algo na tela, que conta enquanto muda o "mundo", ou poderia alterar o 
+estado interno da máquina de uma forma que irá afetar as declarações que vierem depois dela. 
+Essas mudanças são chamadas de _((efeito(s) colateral))is_. 
+As declarações no exemplo anterior apenas produzem os valores `1` e `true` e, então, imediatamente, os joga fora. 
```
imediatamente, -> imediatamente

joaostein commented on 02_program_structure.md:
```diff
@@ -132,15 +122,15 @@ console.log(mood);
-reference to. When you need to remember something, you grow a tentacle
-to hold on to it or you reattach one of your existing tentacles to it.
+Você deveria imaginar ligações como tentáculos, ao invés de caixas.
+Elas não _contêm_ valores, elas _pegam_ eles, duas ligações podem se referir ao
```
`, duas ligações podem se referir ao mesmo valor.` -> 
`(duas ligações podem se referir ao mesmo valor).`

colocar entre parênteses ou colocar em uma nova frase.

joaostein commented on 02_program_structure.md:
````diff
@@ -457,23 +435,21 @@ if (num < 10) {
 }
 ```
+O programa vai primeiro checar se `num` é menor que 10. Se for,
+ele escolherá essa ramificação, mostre "Small" e pronto. Se não for,
````
mostre "Small" -> mostrando `` `"Small"` ``

joaostein commented on 02_program_structure.md:
```diff
@@ -457,23 +435,21 @@ if (num < 10) {
+O programa vai primeiro checar se `num` é menor que 10. Se for,
+ele escolherá essa ramificação, mostre "Small" e pronto. Se não for,
+ele pegará a ramificação do `else`, que contém um segundo `if`. Se a segunda
+condição (`< 100`) for satisfeita, isso significa que o número está entre 10 e 100, e "Medium" é exibido.
```
`100, e` -> `100 e`
"Medium" -> `` `"Medium"` ``

joaostein commented on 02_program_structure.md:
```diff
@@ -572,30 +542,29 @@ console.log(yourName);
+pode se tonar difícil de perceber onde um bloco termina e o outro começa.
+Com a indentação apropriada, a forma visual de um programa corresponde à forma dos blocos dentro dele.
+Eu gosto de usar dois espaços para cada bloco aberto, mas gostos diferem. Algumas pessoas usam
+quatro espaços, e outras usam o ((caractere tab)). O mais importante é que cada novo bloco adicione
```
`espaços, e outras` - > `espaços e outras`

joaostein commented on 02_program_structure.md:
```diff
@@ -790,25 +751,22 @@ switch (prompt("What is the weather like?")) {
-executed.
+Você poderia inserir quantos rótulos de `case` quiser dentro de um bloco aberto por `switch`.  
+O programa iniciará a execução no rótulo que corresponde ao valor que foi dado ao `switch`,
+ou pelo `default` se nenhuma combinação de valores for encontrada. Ele irá continuar executando,
```
irá continuar -> continuará

joaostein reviewed (changes requested):
Quarta leva de reviews. Acho que depois dessa provavelmente estará pronto para merge.

GSFZamai committed:
Finalizado revisão

Finalizado a quarta revisão

joaostein commented:
Valeu, @GSFZamai 🎉
